app/uk/gov/hmrc/tariffclassificationfrontend/service/EventsService.scala

```diff
@@ -32,4 +34,13 @@ class EventsService @Inject()(connector: BindingTariffClassificationConnector) {
     connector.findEvents(reference)
   }
 
+  def addNote(reference: String, note: String, operator: Operator, clock: Clock = Clock.systemDefaultZone())(implicit hc: HeaderCarrier): Future[Unit] = {
+    val event = NewEventRequest(Note(Some(note)), operator, ZonedDateTime.now(clock))
+    connector.createEvent(reference, event)
+      .recover {
+        case t: Throwable => Logger.error(s"Could not create Event for case [$reference] with payload [$event]", t)
+      }
+      .map(_ => ())
```

Comment:
Why are you mapping to `()` ?
I was wondering if it could be more correct to return here a `Future[Event]`  - as it comes from the connector.
Then in the controller you read the JSON response and display the fields you need in the front-end...

Reply:
I was following the pattern from CasesService however I agree that the services should return the created item from the connector.

In this specific case we do a redirect after POST so the controller doesn't need the Event.

Reply:
Fixed